src/extension.ts

```diff
@@ -95,27 +94,39 @@ export function activate(context: vscode.ExtensionContext) {
             p => {
               return new Promise((resolve, reject) => {
                 p.report({ message: respData.data });
-                setTimeout(function() {
-                  resolve();
-                  if (
-                    respData &&
-                    respData.hasOwnProperty('diagCount') &&
-                    vscode.window.activeTextEditor &&
-                    ((respData.diagCount > 0 &&
-                      respData.diagCount !== diagCountInfo) ||
-                      !onFileOpen ||
-                      (onFileOpen &&
-                        onFileOpen.indexOf(
-                          vscode.window.activeTextEditor.document.fileName
-                        ) === -1))
-                  ) {
-                    diagCountInfo = respData.diagCount;
-                    onFileOpen.push(
-                      vscode.window.activeTextEditor.document.fileName
-                    );
-                    showInfoOnfileOpen(respData.data);
-                  }
-                }, 2500);
+                if (
+                  respData &&
+                  respData.hasOwnProperty('diagCount') &&
+                  vscode.window.activeTextEditor &&
+                  ((respData.diagCount > 0 &&
+                    respData.diagCount !== diagCountInfo) ||
+                    !onFileOpen ||
+                    (onFileOpen &&
+                      onFileOpen.indexOf(
+                        vscode.window.activeTextEditor.document.fileName
+                      ) === -1))
+                ) {
+                  setTimeout(() => {

```

Comment:
why are you using setTimeout?

Reply:
`setTimeout` is used here to resolve the status after 1.5 secs as in this case notification are coming via LS based on various states